packages/vanilla/src/components/button/button.html

```diff
@@ -1,4 +1,4 @@
-<a tabindex="0" class="hig__button">
+<a {{#link}}href={{link}}{{/link}} tabindex="0" class="hig__button">
```

Comment:
Is this effectively different from just excluding the `href` attribute in the template, and having it set in JS? Just wondering since that was the approach I took recently, and perhaps those need to be updated.

Reply:
The template is only rendered once when a component is instantiated so this `{{#link}}href={{link}}{{/link}}` only helps if you're instantiating a component with a default like this:

```
const myButton = new Button({ title: "Example", link: "http://example.com" }).
```

Calling `setLink` in a component's `componentDidMount` callback is equally effective.

```
class Button extends Core {
  constructor(options) {
    super(options);
    this._render(Template, options);
    this.initialOptions = options;
  }

  _componentDidMount() {
    if (this.initialOptions.link) {
      this.setLink(this.initialOptions.link);
    }
}
```

Reply:
To close the thread: I added the conditional hrefs just to continue the accepted pattern, even if they may not be applicable in specific cases.